src/catalog/src/system_schema/pg_catalog.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod pg_catalog_memory_table;
+mod table_names;
+
+use std::collections::HashMap;
+use std::sync::{Arc, Weak};
+
+use common_catalog::consts::{self, PG_CATALOG_NAME};
+use datatypes::schema::ColumnSchema;
+use lazy_static::lazy_static;
+use paste::paste;
+use pg_catalog_memory_table::get_schema_columns;
+use table::TableRef;
+pub use table_names::*;
+
+use super::memory_table::tables::u32_column;
+use super::memory_table::MemoryTable;
+use super::{SystemSchemaProvider, SystemSchemaProviderInner, SystemTableRef};
+use crate::CatalogManager;
+
+lazy_static! {
+    static ref MEMORY_TABLES: &'static [&'static str] = &[table_names::PG_TYPE];
+}
+
+/// The column name for the OID column.
+/// The OID column is a unique identifier of type u32 for each object in the database.
+const OID_COLUMN_NAME: &str = "oid";
+
+fn oid_column() -> ColumnSchema {
+    u32_column(OID_COLUMN_NAME)
+}
+
+/// [`PGCatalogProvider`] is the provider for a schema named `pg_catalog`, it is not a catalog.
+pub struct PGCatalogProvider {
+    catalog_name: String,
+    _catalog_manager: Weak<dyn CatalogManager>,
+    tables: HashMap<String, TableRef>,
+}
+
+impl SystemSchemaProvider for PGCatalogProvider {
+    fn tables(&self) -> &HashMap<String, TableRef> {
+        assert!(!self.tables.is_empty());
+
+        &self.tables
+    }
+}
+
+// TODO(j0hn50n133): Not sure whether to avoid duplication with `information_schema` or not.
```

Comment:
We can first implement the feature, and then do a clean up. Otherwise the PR will be mixed and hard to review.

Keep this TODO is good.